### PR TITLE
agent dns: optimize cname resolution

### PR DESCRIPTION
**Pull request by rshriram**
Chain the cname and the resolved A/AAAA record in same response
to keep curl and friends happy.

Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[x] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[x] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: pilot/pkg/dns/dns.go [hunk 1]
```diff
@@ -224,40 +235,64 @@ func separateIPtypes(ips []string) (ipv4, ipv6 []net.IP) {
...
+		hostname = cn[0].(*dns.CNAME).Target
 	}
-	return ans
+	var ipAnswers []dns.RR
+	switch qtype {
+	case dns.TypeA:
+		ipAnswers = table.name4[hostname]
+	case dns.TypeAAAA:
+		ipAnswers = table.name6[hostname]
+		// TODO: handle PTR records for reverse dns lookups
```
Comment (howardjohn) [inline pilot/pkg/dns/dns.go:283]: if its a different type, should we return `nil, false` so we always query upstream?

Comment (howardjohn) [reply pilot/pkg/dns/dns.go:283]: Otherwise we drop all other query types if the host is found right?

**Review by howardjohn — APPROVED** [review]

File: pilot/pkg/dns/dns.go (hunk 1 above)
Comment (rshriram) [reply pilot/pkg/dns/dns.go:283]: ah true

**Pull request merged**